checks/binary_artifact.go

```diff
@@ -24,19 +24,22 @@ import (
 // CheckBinaryArtifacts is the exported name for Binary-Artifacts check.
 const CheckBinaryArtifacts string = "Binary-Artifacts"
 
-//nolint
+// nolint
 func init() {
 	supportedRequestTypes := []checker.RequestType{
 		checker.CommitBased,
 	}
-	if err := registerCheck(CheckBinaryArtifacts, BinaryArtifacts, supportedRequestTypes); err != nil {
+	b := BinaryArtifactsCheck{}
+	if err := registerCheckInterface(CheckBinaryArtifacts, b, supportedRequestTypes); err != nil {
 		// this should never happen
 		panic(err)
 	}
 }
 
-// BinaryArtifacts  will check the repository contains binary artifacts.
-func BinaryArtifacts(c *checker.CheckRequest) checker.CheckResult {
```

Comment:
Can we do this in a way that doesn't remove this signature (or at least yet)? I know Allstar calls the BinaryArtifacts function, and wouldn't be surprised if others do as well.

Then when we bump the major version we can make the change?

Reply:
Ok, we can do that.

Reply:
👍